woocommerce/woocommerce-android
# [Custom Fields] Show an entry point in product details screen

Closes: #12677 


### Description
This PR adds logic to show an entry point to the custom fields screen from the product details, the entry point will be shown either as a row if the product had already some custom fields, or in the details bottom sheet when it doesn't.

**Note:** We don't have an empty view for the custom fields screen yet, it will be added later.

### Steps to reproduce
1. Open the app.
2. Open a product details.

### Testing information
- Confirm that the custom fields row is shown when the product has already some fields.
- Confirm that the custom fields button is shown in the details bottom sheet when no custom fields were added yet.
- Test the navigation logic for both cases.
- Make changes to the custom fields and confirm the logic is reflected.

### The tests that have been performed


### Images/gif
| Product with existing custom fields | Product with no custom fields |
| --- | --- |
| ![Screenshot_20240925_092727](https://github.com/user-attachments/assets/fb6adae4-0cb3-4704-a981-9634e23e3dcb) | ![Screenshot_20240925_092737](https://github.com/user-attachments/assets/7237ef4f-7ca5-4dae-b87c-bccd5d274e29) |



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
